pkg/receive/handler.go

```diff
@@ -482,6 +548,62 @@ func (h *Handler) receiveHTTP(w http.ResponseWriter, r *http.Request) {
 	h.writeSamplesTotal.WithLabelValues(strconv.Itoa(responseStatusCode), tenant).Observe(float64(totalSamples))
 }
 
+// QueryMetaMonitoring queries any Prometheus Query API compatible meta-monitoring
+// solution with the configured query for getting current HEAD series of all tenants.
+// It then populates tenantCurrentSeries map with result.
+func (h *Handler) QueryMetaMonitoring(ctx context.Context) error {
+	c := promclient.NewWithTracingClient(h.logger, h.metaMonitoringClient, httpconfig.ThanosUserAgent)
+
+	vectorRes, _, err := c.QueryInstant(ctx, h.options.MetaMonitoringUrl, h.options.MetaMonitoringLimitQuery, time.Now(), promclient.QueryOptions{})
+	if err != nil {
+		h.metaMonitoringErr.Inc()
+		return err
+	}
+
+	level.Debug(h.logger).Log("msg", "successfully queried meta-monitoring", "vectors", len(vectorRes))
+
+	// Construct map of tenant name and current HEAD series.
```

Comment:
```suggestion
	// Construct map of tenant name and current active series.
```

Uppercase HEAD looks like HTTP HEAD method - let's avoid this confusion.

Reply:
Changed this in https://github.com/thanos-io/thanos/pull/5520/commits/a8449564cda0fd8e139a9e7443090dfcdb443fdb